src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -248,25 +248,16 @@ private bool TypeIsCollectionOfNumbersWithInternalConverter()
             Type? elementType = ConverterBase.ElementType;
             Debug.Assert(elementType != null);
 
-            elementType = Nullable.GetUnderlyingType(elementType) ?? elementType;
-
-            if (elementType == typeof(byte) ||
-                elementType == typeof(decimal) ||
-                elementType == typeof(double) ||
-                elementType == typeof(short) ||
-                elementType == typeof(int) ||
-                elementType == typeof(long) ||
-                elementType == typeof(sbyte) ||
-                elementType == typeof(float) ||
-                elementType == typeof(ushort) ||
-                elementType == typeof(uint) ||
-                elementType == typeof(ulong) ||
-                elementType == JsonClassInfo.ObjectType)
+            try
             {
-                return true;
+                JsonConverter converter = Options.GetConverter(elementType);
+                return converter.IsInternalConverterForNumberType;
+            }
+            catch (NotSupportedException)
```

Comment:
I'm inclined to just call the custom converter and if it doesn't know about quoted numbers, then let it throw what it would have otherwise (likely a reader exception re-thrown as JsonException).

Reply:
Calling the custom converter is fine with me.

Reply:
> I'm inclined to just call the custom converter and if it doesn't know about quoted numbers, then let it throw what it would have otherwise (likely a reader exception re-thrown as JsonException).

Why? I haven't fully thought through the scenario or use case, so _definitely not pushing back_, but just curious what is the benefit to the caller? Could such a behavior be confusing or would it match their expectation?

Reply:
Generally, the expectation when users provide custom converters is that they are called. The exception is when `JsonIgnoreCondition.WhenWritingNull/Default`/`IgnoreNullValues` is active and the value is `null`/`default`.

The initial concern was that it didn't make sense to specify both custom handling and a custom converter for a type/property since the custom handling would not be honored (by the serializer). Thus, logic was added to throw when both custom handling and a custom handling were provided. The fix would be to remove either the custom number handling or the custom converter. The problem with this is highlighted in https://github.com/dotnet/runtime/pull/42239#issuecomment-695371432.

> The current restriction makes it impossible to have a custom converter for a property on a non-owned type which has number handling specified

The downside to removing the restriction on custom converters is that users can now specify an option which will not be honored. The upside is that we maintain the behavior that a custom converter can take over the (de)serialization of any type, even if parts of the object graph is non-owned.

Reply:
That makes sense.

Have we gotten any customer feedback around this, to validate that the current restriction is problematic?

What is the principles behind when an option is allowed to coincide with converters, and overrides its behavior, and when is it not allowed or ignored, with the converter being prioritized?

I think it will be valuable to enumerate the design and precedence more broadly, across all options and permutations with converters.